src/sidebar/util/term-before-position.ts

```diff
@@ -28,16 +26,20 @@ export function getContainingWordOffsets(
   text: string,
   referencePosition: number,
 ): WordOffsets {
-  const precedingEmptyCharPos = text
-    .slice(0, referencePosition)
-    .search(/\s\S*$/);
-  const subsequentEmptyCharPos = text.slice(referencePosition).search(/\s/);
+  const precedingText = text.slice(0, referencePosition);
+  const matches = [...precedingText.matchAll(/[\s,.;:|?!'"\-()[\]{}]/g)];
+  const precedingCharPos =
+    matches.length > 0 ? Math.max(...matches.map(match => match.index)) : -1;
+
+  const subsequentCharPos = text
+    .slice(referencePosition)
+    .search(/[\s,.;:|?!'"\-()[\]{}]/);
 
   return {
-    start: (precedingEmptyCharPos === -1 ? -1 : precedingEmptyCharPos) + 1,
+    start: precedingCharPos + 1,
```

Comment:
😅 

This was similar to

```js
const something = returnsBoolean();
if (something) {
  return true;
} else {
  return something.
}
```

My IDE usually pops a warning saying "this can be simplified", but it didn't here for some reason.